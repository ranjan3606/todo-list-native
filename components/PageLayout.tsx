import React from 'react';
import { 
  View, 
  StyleSheet, 
  SafeAreaView, 
  Platform, 
  StatusBar,
  Text,
  ViewStyle,
  TextStyle
} from 'react-native';
import { Colors } from '@/constants/Colors';
import { useActualColorScheme } from '@/services/theme';

type PageLayoutProps = {
  children: React.ReactNode;
  title?: string;
  style?: ViewStyle;
  titleStyle?: TextStyle;
  showHeader?: boolean;
};

export function PageLayout({ 
  children, 
  title,
  style,
  titleStyle,
  showHeader = true
}: PageLayoutProps) {
  const colorScheme = useActualColorScheme();

  return (
    <SafeAreaView style={[
      styles.safeArea, 
      { backgroundColor: Colors[colorScheme].background }
    ]}>
      <View style={[styles.container, style]}>
        {showHeader && title && (
          <Text style={[
            styles.header, 
            { color: Colors[colorScheme].text },
            titleStyle
          ]}>
            {title}
          </Text>
        )}
        {children}
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
    paddingTop: Platform.OS === 'android' ? StatusBar.currentHeight : 0,
  },
  container: {
    flex: 1,
    padding: 16,
    paddingBottom: Platform.OS === 'ios' ? 80 : 70, // Account for tab bar height
  },
  header: {
    fontSize: 24,
    fontWeight: 'bold',
    marginBottom: 8,
    marginTop: 8,
  },
});
