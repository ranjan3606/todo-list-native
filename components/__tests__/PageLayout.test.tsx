import React from 'react';
import { render } from '@testing-library/react-native';
import { PageLayout } from '../PageLayout';

jest.mock('@/constants/Colors', () => ({
  Colors: {
    light: { background: '#fff', text: '#000' },
    dark: { background: '#000', text: '#fff' },
  }
}));

jest.mock('@/services/theme', () => ({
  useActualColorScheme: () => 'light',
}));

describe('PageLayout', () => {
  it('renders correctly (snapshot)', () => {
    const { toJSON } = render(
      <PageLayout title="Test Title">
        <React.Fragment></React.Fragment>
      </PageLayout>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders without header', () => {
    const { toJSON } = render(
      <PageLayout showHeader={false}>
        <React.Fragment></React.Fragment>
      </PageLayout>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('applies custom style and titleStyle', () => {
    const { getByText } = render(
      <PageLayout title="Styled Title" style={{ backgroundColor: 'red' }} titleStyle={{ color: 'blue' }}>
        <></>
      </PageLayout>
    );
    expect(getByText('Styled Title')).toBeTruthy();
  });
});
