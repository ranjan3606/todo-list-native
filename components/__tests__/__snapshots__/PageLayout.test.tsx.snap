// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders correctly (snapshot) 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
        "paddingTop": 0,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "padding": 16,
          "paddingBottom": 80,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "fontSize": 24,
            "fontWeight": "bold",
            "marginBottom": 8,
            "marginTop": 8,
          },
          {
            "color": "#000",
          },
          undefined,
        ]
      }
    >
      Test Title
    </Text>
  </View>
</RCTSafeAreaView>
`;

exports[`PageLayout renders without header 1`] = `
<RCTSafeAreaView
  style={
    [
      {
        "flex": 1,
        "paddingTop": 0,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
          "padding": 16,
          "paddingBottom": 80,
        },
        undefined,
      ]
    }
  />
</RCTSafeAreaView>
`;
